lib/containers/common.pm

```diff
@@ -25,13 +26,17 @@ sub is_unreleased_sle {
 }
 
 sub activate_containers_module {
-    my ($running_version, $sp, $host_distri) = get_os_release;
-    my $suseconnect = script_output_retry('SUSEConnect --status-text', timeout => 240, retry => 3, delay => 60);
-    if ($suseconnect !~ m/Containers/) {
-        $running_version eq '12' ? add_suseconnect_product("sle-module-containers", 12) : add_suseconnect_product("sle-module-containers");
-        $suseconnect = script_output_retry('SUSEConnect --status-text', timeout => 240, retry => 3, delay => 60);
+    my $json = Mojo::JSON::decode_json(script_output_retry('SUSEConnect -s', timeout => 240, retry => 3, delay => 60));
+    my ($not_registered, $version, $arch);
+    foreach (@$json) {
+        if ($_->{identifier} =~ 'container' && $_->{status} =~ 'Not Registered') {
+            $not_registered = 1;
+            $version = $_->{version};
+            $arch = $_->{arch};
+        }
```

Comment:
I would add a `last` here, since at this point there is no need to further process the other json elements.